lib/Backends/OpenCL/OpenCLDeviceManager.cpp

```diff
@@ -62,17 +62,52 @@ cl_mem OpenCLDeviceManager::allocDeviceBuffer(uint64_t size) {
 OpenCLDeviceManager::OpenCLDeviceManager(std::unique_ptr<DeviceConfig> config)
     : QueueBackedDeviceManager(BackendKind::OpenCL, std::move(config)) {}
 
+llvm::Error OpenCLDeviceManager::parseConfig() {
+  auto it = config_->parameters.find("deviceId");
+  if (it != config_->parameters.end()) {
+    char *end;
+    auto input = it->second.c_str();
+    auto parsed = strtol(input, &end, 10);
+    if (end == input) {
```

Comment:
This seems like an incorrect error condition check. E.g. you could pass in "1a2" and it wouldn't catch the error, right? I think you want to check that end got to the end of input, not that it's stuck at the beginning of the input.

Reply:
Also this seems like a good place to use a helper, since it's also used below for `platformId` and probably will be used across other DeviceManagers as well?